src/Components/ADT3DSceneBuilder/Internal/Behaviors/VisualRules/VisualRuleList.tsx

```diff
@@ -1,7 +1,75 @@
-import React from 'react';
-import { IVisualRulesListProps } from './VisualRules.types';
+import { IContextualMenuItem } from '@fluentui/react';
+import React, { useEffect, useState } from 'react';
+import { TFunction, useTranslation } from 'react-i18next';
+import { CardboardList } from '../../../../CardboardList';
+import { ICardboardListItem } from '../../../../CardboardList/CardboardList.types';
+import { IVisualRule, IVisualRulesListProps } from './VisualRules.types';
+/**
+ *
+ * Visual Rule List will handle the generation of ruleItems and the actions on a rule
+ */
 
-export const VisualRulesList: React.FC<IVisualRulesListProps> = (_props) => {
-    // TODO: Include Cardboard list here
-    return <p>CardboardList here</p>;
+export const VisualRulesList: React.FC<IVisualRulesListProps> = ({
+    ruleItems,
+    onEditRule,
+    onRemoveRule
+}) => {
+    const { t } = useTranslation();
+    //list of data in carbboardlist shape
+    const [listItems, setListItems] = useState<
+        ICardboardListItem<IVisualRule>[]
+    >([]);
+
+    //making sure to display the correct listItem when one of the dependencies below changes
+    useEffect(() => {
+        //making sure to get the rules in cardboardlist data shape
+        const listItems = getListItems(ruleItems, onRemoveRule, onEditRule, t);
+        setListItems(listItems);
+    }, [ruleItems, onRemoveRule, onEditRule, t]);
+
+    return (
+        <CardboardList<IVisualRule>
+            listKey={'visualRules-in-behavior'}
+            items={listItems}
+        />
+    );
 };
+
+function getListItems(
+    rules: IVisualRule[],
+    onRemoveRule: (ruleItem: string) => void,
+    onEditRule: (ruleItem: string) => void,
+    t: TFunction<string>
+) {
+    const getMenuItems = (item: IVisualRule): IContextualMenuItem[] => {
+        return [
+            {
+                key: 'edit',
+                'data-testid': 'editRuleOverflow',
+                text: t('3dSceneBuilder.behaviorVisualRulesForm.editRule'),
+                iconProps: { iconName: 'Edit' },
+                onClick: () => onEditRule(item.id)
+            },
+            {
+                key: 'remove',
+                'data-testid': 'removeRuleOverflow',
+                text: t('3dSceneBuilder.behaviorVisualRulesForm.removeRule'),
+                iconProps: { iconName: 'Delete' },
+                onClick: () => onRemoveRule(item.id)
+            }
+        ];
+    };
+
+    return rules.map((item) => {
+        const viewModel: ICardboardListItem<IVisualRule> = {
+            ariaLabel: '',
+            iconStart: { name: '' },
+            item: item,
+            openMenuOnClick: true,
+            overflowMenuItems: getMenuItems(item),
+            textPrimary: item.displayName,
+            textSecondary: item.conditions.join(', ')
```

Comment:
does this align to the latest designs? I think we decided to list out the number of conditions or something? @pasanchMSFT do you recall the final decision?

Reply:
Seems like it matches the mock data, but when real data comes in this will have to change @abalogun-dev. This should actually show the number of badges and mesh colorings. i.e. `x badges, y mesh colorings`. Can be done when wired to real data.

Reply:
imo, this can change in next PR.

Reply:
yup, it can. That'll be in the next PR